File: backend/app/api/schemas/imports.py
from pydantic import BaseModel


class ImportUploadResponse(BaseModel):
    import_id: str
    headers: list[str]
    column_map: dict[str, str]
    encoding: str
    delimiter: str
    total_rows: int
    preview: list[dict]


class ImportConfirmRequest(BaseModel):
    column_map: dict[str, str]
    account_name: str = "Основной счёт"
    date_format: str = "%Y-%m-%d"


class ImportConfirmResponse(BaseModel):
    import_id: str
    status: str
    imported_rows: int
    skipped_rows: int
    duplicate_rows: int

File: frontend/src/shared/api/imports.ts
import { api } from "./client"

export interface ImportUploadResponse {
  import_id: string
  headers: string[]
  column_map: Record<string, string>
  encoding: string
  delimiter: string
  total_rows: number
  preview: Record<string, unknown>[]
}

export interface ImportConfirmRequest {
  column_map: Record<string, string>
  account_name?: string
  date_format?: string
}

export interface ImportConfirmResponse {
  import_id: string
  status: string
  imported_rows: number
  skipped_rows: number
  duplicate_rows: number
}

export const importsApi = {
  upload: (file: File) => {
    const form = new FormData()
    form.append("file", file)
    return fetch("/api/v1/imports/upload", {
      method: "POST",
      body: form,
    }).then((r) => {
      if (!r.ok) throw new Error(r.statusText)
      return r.json() as Promise<ImportUploadResponse>
    })
  },
  confirm: (importId: string, body: ImportConfirmRequest) =>
    api.post<ImportConfirmResponse>(`/imports/${importId}/confirm`, body),
  get: (importId: string) => api.get<{ id: string; status: string; imported_rows: number }>(`/imports/${importId}`),
}
